openstack_dashboard/api/glance.py

```diff
@@ -153,23 +157,25 @@ def __ne__(self, other_image):
         return not self.__eq__(other_image)
 
     def get_catalog_id(self):
-        if Image.PUBLISHED_APPLIANCES.get(self.id):
-            return Image.PUBLISHED_APPLIANCES.get(self.id).get('id', -1)
+        if cache.get('published_appliances', {}).get(self.id):
+            return cache.get('published_appliances', {}).get(self.id).get('id', -1)
         return -1
 
     def get_is_project_supported(self):
-        if Image.PUBLISHED_APPLIANCES.get(self.id):
-            return Image.PUBLISHED_APPLIANCES.get(self.id).get('project_supported', False)
+        if cache.get('published_appliances', {}).get(self.id):
+            return cache.get('published_appliances', {}).get(self.id).get('project_supported', False)
         return False
 
     def get_is_published_in_app_catalog(self):
-        return Image.PUBLISHED_APPLIANCES.get(self.id) is not None
+        return cache.get('published_appliances', {}).get(self.id) is not None
 
 def fetch_published_appliances():
-    if not cache.get('app_catalog_updated'):
+    if not cache.get('app_catalog_updated', False) or not cache.get('published_appliances', False):
```

Comment:
I don't think you need to store the 'updated' flag if you are doing this solution. This pattern you are implementing is a [cache-aside](https://blog.cdemi.io/design-patterns-cache-aside-pattern/) pattern. You can express it a bit simpler by just doing the following:

```python
appliances = cache.get('published_appliances')

if not appliances:
  appliance_list = requests.get(settings.CHAMELEON_PORTAL_API_BASE_URL\
    + settings.APPLIANCE_CATALOG_API_PATH).json().get('result')
  appliances = {}
  for appliance in appliance_list:
    # processing...
  cache.set('published_appliances', appliances, 60*15)

# Now, `appliances` should exist AND be in cache
return appliances
```

Reply:
(keeping the `try`/`except` that you have, of course!)

Reply:
Yes, thanks! I'm familiar with the pattern, I've worked with it a few times over the years :). I kept the cache timeout here so other requests could still use the existing cache--in the event that the process of requesting and generating a new dict fails or happens to take a long time. But, I admit it's probably not necessary, I'll update the code to pull that part out, thanks!

Reply:
OK, I can see that you thought a lot about it. I'm biased towards the simpler solution of one key, but your point is very valid.